server/src/main/java/org/elasticsearch/index/search/QueryParserHelper.java

```diff
@@ -133,6 +134,11 @@ private QueryParserHelper() {}
                 continue;
             }
 
+            if (wildcardMatch && fieldType.includeInFieldExpansion(context) == false) {
+                // Ignore fields in the mappings that are not present in the lucene index
+                continue;
+            }
```

Comment:
Is there a reason why we should only do this when `wildcardMatch` is true? Is it so that we propagate exceptions when querying illegal fields? If this is the case, then I think we should properly check whether `fieldOrPattern` is a pattern since patterns could resolve to a single field in some cases.

Reply:
I think we could reuse `acceptAllTypes` here? It seems to represents whether the pattern contained a wildcard. Then we could combine the check with `fieldType.getTextSearchInfo() == TextSearchInfo.NONE`.

Reply:
Reusing `acceptAllTypes` works nicely here, good spot.